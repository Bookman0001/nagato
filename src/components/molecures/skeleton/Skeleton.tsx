import styled, { css, keyframes } from 'styled-components'

export function Skeleton() {
  return (
    <Wrapper>
      <SubTitleSkeleton data-testid={'title'} />
      <DateSkeleton data-testid={'date'} />
    </Wrapper>
  )
}

const Wrapper = styled.div`
  display: flex;
  padding-bottom: 4rem;
`

const Animation = keyframes`
  0% {
    transform: translateX(-100%);
  }
  100% {
    transform: translateX(100%);
  }
`

const skeleton = css`
  height: 30px;
  background-color: #dcdcdc;
  position: relative;
  overflow: hidden;
  @media (prefers-color-scheme: dark) {
    background-color: #696969;
  }
  ::before {
    content: '';
    display: block;
    height: 100%;
    width: 100%;
    background: linear-gradient(
      90deg,
      transparent,
      rgba(255, 255, 255, 0.5),
      transparent
    );
    position: absolute;
    top: 0;
    left: 0;
    animation: ${Animation} 1.2s linear infinite;
  }
`

const SubTitleSkeleton = styled.div`
  width: 100%;
  margin-right: auto;
  ${skeleton}
`

const DateSkeleton = styled.div`
  width: 13rem;
  margin-left: 2rem;
  ${skeleton}
`
