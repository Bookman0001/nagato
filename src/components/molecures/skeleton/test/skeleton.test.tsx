/**
 * @jest-environment jsdom
 */
import { shallow } from 'enzyme'

import Skeleton from '../'

test('Skeleton', () => {
  const component = shallow(<Skeleton />)
  expect(component).toMatchSnapshot()
})
