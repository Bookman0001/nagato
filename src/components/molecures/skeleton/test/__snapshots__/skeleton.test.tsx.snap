// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Skeleton 1`] = `
<styled.div>
  <styled.div />
  <styled.div>
    ..........
  </styled.div>
</styled.div>
`;
